pkgs/development/compilers/gnatboot/default.nix

```diff
@@ -1,55 +1,106 @@
-{ lib, stdenv, fetchurl }:
-
-if stdenv.hostPlatform != stdenv.targetPlatform
-then builtins.throw "gnatboot can't cross-compile"
-else
-
-stdenv.mkDerivation {
-  pname = "gentoo-gnatboot";
-  version = "4.1";
-
-  src = if stdenv.hostPlatform.system == "i686-linux" then
-    fetchurl {
-      url = "mirror://gentoo/distfiles/gnatboot-4.1-i386.tar.bz2";
-      sha256 = "0665zk71598204bf521vw68i5y6ccqarq9fcxsqp7ccgycb4lysr";
-    }
-  else if stdenv.hostPlatform.system == "x86_64-linux" then
-    fetchurl {
-      url = "mirror://gentoo/distfiles/gnatboot-4.1-amd64.tar.bz2";
-      sha256 = "1li4d52lmbnfs6llcshlbqyik2q2q4bvpir0f7n38nagp0h6j0d4";
-    }
-  else
-    throw "Platform not supported";
-
-  dontStrip = 1;
+{ stdenv, lib, autoPatchelfHook, fetchzip, lzma, ncurses5, readline, gmp, mpfr
+, expat, libipt, zlib, dejagnu, sourceHighlight, python3, elfutils, guile, glibc
+}:
 
-  installPhase = ''
-    mkdir -p $out
-    cp -R * $out
+stdenv.mkDerivation rec {
+  pname = "gnatboot";
+  version = "11.2.0-4";
+  src = fetchzip {
+    url =
+      "https://github.com/alire-project/GNAT-FSF-builds/releases/download/gnat-"
+      + version + "/gnat-x86_64-linux-" + version + ".tar.gz";
+    hash = "sha256-8fMBJp6igH+Md5jE4LMubDmC4GLt4A+bZG/Xcz2LAJQ=";
+  };
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    lzma
+    ncurses5
+    readline
+    gmp
+    mpfr
+    expat
+    libipt
+    zlib
+    dejagnu
+    sourceHighlight
+    python3
+    elfutils
+    guile
+    glibc
+  ];
+
+  propagatedNativeBuildInputs = [
+    autoPatchelfHook
+    lzma
+    ncurses5
+    readline
+    gmp
+    mpfr
+    expat
+    libipt
+    zlib
+    dejagnu
+    sourceHighlight
+    python3
+    elfutils
+    guile
+    glibc
+  ];
```

Comment:
Please properly sort the dependencies. Having all of them in all inputs is most likely not the minimal dependencies.

Reply:
I'll sort them, but I'm pretty sure all of them are required.

Reply:
Got it. Thanks.